F21/yzhang120/lower_case.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+
+def lower_general(input_string):
+    return input_string.lower()
+
+def lower_null(input_string):
+    return input_string
+
+def lower_tr_az(input_string):
+    output_string = ''
+    for char in input_string:
+        if char != 'I':
+            output_string += char.lower()
+        else:
+            output_string += u"\u0131"
+    return output_string
+
+def lower_ga(input_string):
```

Comment:
While this code should work in theory but this code may not handle if the unicode for accented letters is decomposed unicode instead of normalized NFC unicode. This may cause your code to incorrectly handle Gaelic text if it has vowels written using two chars (" ‘+o ") instead of one char (" ó ").

Reply:
I would look up unicode normalization.
https://docs.python.org/3/library/unicodedata.html
https://towardsdatascience.com/difference-between-nfd-nfc-nfkd-and-nfkc-explained-with-python-code-e2631f96ae6c